pkg/infra/httpclient/httpclientprovider/http_client_provider.go

```diff
@@ -29,5 +33,23 @@ func New(cfg *setting.Cfg) httpclient.Provider {
 
 	return newProviderFunc(sdkhttpclient.ProviderOptions{
 		Middlewares: middlewares,
+		ConfigureTransport: func(opts sdkhttpclient.Options, transport *http.Transport) {
+			newConntrackRoundTripper(connName)
+		},
 	})
 }
+
+// newConntrackRoundTripper takes a http.DefaultTransport and adds the Conntrack Dialer
+// so we can instrument outbound connections
+func newConntrackRoundTripper(name string) *http.Transport {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.DialContext = conntrack.NewDialContextFunc(
+		conntrack.DialWithTracing(),
+		conntrack.DialWithName(name),
+		conntrack.DialWithDialer(&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
```

Comment:
Should take the created transport and use that, i.e. `conntrack.DialWithDialer(transport.Dial...)`